lib/galaxy/webapps/galaxy/controllers/visualization.py

```diff
@@ -1015,6 +1017,45 @@ def phyloviz( self, trans, id=None, dataset_id=None, tree_index=0, **kwargs ):
             }
         return trans.fill_template_mako( "visualization/phyloviz.mako", data=data, config=config )
 
+    @web.expose
+    @web.require_login( "run Galaxy Interactive Environments" )
+    def gie_list( self, trans, **kwargs ):
+        if not hasattr( self, 'gie_image_map' ):
+            self.gie_image_map = {}
+
+            for gie_dir in self.app.config.gie_dirs:
+                gie_list = os.listdir( gie_dir )
+                for gie in gie_list:
+                    gie_path = os.path.join(gie_dir, gie)
+
+                    if not os.path.isdir(gie_path):
+                        continue
+
+                    if not os.path.exists(self._gie_config_dir(gie_path)):
+                        continue
+
+                    if os.path.exists( self._gie_config_dir( gie_path, 'allowed_images.yml' ) ):
+                        image_file = self._gie_config_dir( gie_path, 'allowed_images.yml' )
+                    elif os.path.exists( self._gie_config_dir( gie_path, 'allowed_images.yml.sample' ) ):
+                        image_file = self._gie_config_dir( gie_path, 'allowed_images.yml.sample' )
+                    else:
+                        continue
+
+                    with open( image_file, 'r' ) as handle:
```

Comment:
There should be away to hide IE's from this list. For example BAM.iobio is more of a visualisation and should probably not listed there. If we want to list all of them in this box we need to implement file-type filtering and so on ... I would strictly annotate IE's that should be displayed in the top-bar. Anything else is only available with the VIS icon.